app/services/git_based_domain_import_service.rb

```diff
@@ -50,14 +50,14 @@ def queue_refresh(git_repo_id)
     MiqTask.generic_action_with_callback(task_options, queue_options)
   end
 
-  def queue_refresh_and_import(git_url, ref, ref_type, tenant_id)
+  def queue_refresh_and_import(git_url, ref, ref_type, tenant_id, custom_import_options = {})
     import_options = {
       "git_url"   => git_url,
       "ref"       => ref,
       "ref_type"  => ref_type,
       "tenant_id" => tenant_id,
       "overwrite" => true
-    }
+    }.merge(custom_import_options)
```

Comment:
@jvlcek As discussed, this will have to somehow at least know about password and encrypt that onto the queue.  I think it's as simple as `import_options["password"] &&= ManageIQ::Password.try_encrypt(import_options["password"])`

Additionally, you will need to be sure that the keys are strings, so you may want to do `custom_import_options.stringify_keys` somewhere as well.

Reply:
@Fryguy Thank you for the help.  I believe I've addressed the suggestions/issues you raised.